### ISSUE TITLE
Refactor 'keptn upgrade' CLI command code

### ISSUE THREAD

**Issue by odubajDT** [keptn/keptn#6439]
The 'keptn upgrade' CLI command code needs to be refactored to make it more testable. For example `install.go` representing code for CLI command 'keptn install' has an `InstallCmdHandler{}` struct where components for retrieving information from the API can be injected and mocked during the unit tests - the changes in the `upgrade.go` would also benefit from that approach since the `addWarningNonExistingProjectUpstream()` function would not have to initialize the `credentialsManager` and `projectHandler` within the function anymore.

## DoD 

- [ ] Refactor the code to support dependency injection to enable easier mocking of API 
- [ ] Remove the `if mocking` checks
- [ ] Updating the existing tests to properly mock the API endpoints

**Comment by eyskim** [comment]
Hi @odubajDT,

I wanted to attempt this refactoring and have come up with the following approach:
1. Refactor `upgrade.go` and `upgrade_test.go` with a new `UpgradeCmdHandler` struct with dependencies similar to what is used in `install.go`'s `InstallCmdHandler`.
2. Add dependencies to `UpgradeCmdHandler` as required by `addWarningNonExistingProjectUpstream()` and further refactor/add unit tests.

These steps might result in a struct that looks similar to:
```
type UpgradeCmdHandler struct {
	helmHelper        helm.IHelper
	namespaceHandler  kube.IKeptnNamespaceHandler
	userInput         common.IUserInput
        credentialManager credentialmanager.CredentialManager
        projectHandler    api.ProjectsV1Interface
}
```

With the above in mind, I was wondering how I should handle the fact that `ProjectsV1Interface.GetAllProjects()` ([link to code](https://github.com/keptn/keptn/blob/b3e9ceff7355ca8b18432a97b47d524cfee21e84/cli/cmd/upgrade.go#L265-L268)) depends on the prior execution of `CredentialManager.GetCreds(namespace)` ([link to code](https://github.com/keptn/keptn/blob/b3e9ceff7355ca8b18432a97b47d524cfee21e84/cli/cmd/upgrade.go#L253-L256)) to add the auth headers to the `APISet` ([link to code](https://github.com/keptn/keptn/blob/b3e9ceff7355ca8b18432a97b47d524cfee21e84/cli/cmd/upgrade.go#L258-L261)) in `addWarningNonExistingProjectUpstream()`?

In other words, the eventual `projectHandler` will depend on the prior execution of methods available to `credentialManager`.

In this case, I can think of three possible approaches:
1. Inject only `credentialManager` and initialize `projectHandler` from within `addWarningNonExistingProjectUpstream()`
2. Inject only `projectHandler`, assuming the auth headers have been added
5. Inject both `credentialManager` and `projectHandler`, then authenticate `projectHandler` within `addWarningNonExistingProjectUpstream()`. I am least sure about how to implement this approach with the available APIs.

Would love to hear you thoughts about the possible approaches and if there are any other considerations.

Thanks!

**Comment by odubajDT** [comment]
Hello @eyskim ! 

Thank you very much for you interest in contributing to our project. From my perspective the easiest and cleanest way would be to inject `credentialManager` and initialize the `projectHandler` afterwards from where it is needed. Of course, if it is needed only once, then it's enough to do it from `addWarningNonExistingProjectUpstream()`. If you agree, I can assign this ticket to you afterrwards.

Thanks!

**Comment by eyskim** [comment]
Hi @odubajDT, that sounds great! Will get started on this right away!

**Comment by odubajDT** [comment]
Thanks! Assigning the ticket to you, good luck :)

**Comment by thisthat** [comment]
This ticket is superseded by the work done in #7846
